File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/LouseNormal_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class LouseNormal_Model extends MonsterInfo{
    public static Id = "LouseNormal"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(11,16);
        else
            return Mathf.RandomRange(10,15);
    }

    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/louseRed/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/louseRed/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/louseRed/skeleton.png"));
    }

    get Name(): string {
        return "普通型虱虫";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/LouseNormal_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class LouseNormal_Model : MonsterInfo
    {
        public static string Id = "LouseNormal";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(11,16) : Random.Range(10,15);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/louseRed/skeleton.atlas",
                    "Monsters/theBottom/louseRed/skeleton.json", "Monsters/theBottom/louseRed/skeleton.png");
            }
        }

        public override string Name => "普通型虱虫";

        public LouseNormal_Model(Vector2 pos) : base(pos)
        {
        }
    }
}